src/main/scala/nodes/images/Convolver.scala

```diff
@@ -101,29 +101,31 @@ object Convolver {
       whitener: Option[ZCAWhitener],
       varConstant: Double): DenseMatrix[Double] = {
     var x,y,chan,pox,poy,py,px = 0
-    while (x < resWidth) {
-      y = 0
-      while (y < resHeight) {
-        chan = 0
-        while (chan < imgChannels) {
-          poy = 0
-          while (poy < convSize) {
-            pox = 0
-            while (pox < convSize) {
+
+    poy = 0
+    while (poy < convSize) {
```

Comment:
If its something thats used multiple times we can make a utility function out of it ? But it seems like in this particular case there is some interesting logic inside ?

We could go down the macro route, but if simpler things are enough that'd be better